### PR TITLE
Display search chips to the left, to the left

### PR DESCRIPTION
**Pull request by tsop14**
chips

Once created, pushes chips to the right of the search box.
Only  has ONE text chip which stays on the left. 

#1801

### CODE REVIEW

**Comment by NickPapacostas** [comment]
:+1: definitely nicer 

general thoughts:
- min width is better but still gets crowded fast on a small screen (potentially no solution for this)
- colorz! so we can see the chips more clearly all the way over there ^_^

**Comment by ShaunYearStrong** [comment]
I agree with @NickPapacostas regarding colours, it'll mean you don't miss the chip appearing as your focus would be on the cursor at the time.

**Comment by NickPapacostas** [comment]
@ShaunYearStrong  now i feel obligated to mention that the colors were @tsop14 's idea that she mentioned earlier and i unashamedly stole!

**Comment by ShaunYearStrong** [comment]
Okay then, I agree with @tsop14 then, and revoke my support for you!

**Comment by NickPapacostas** [comment]
:ideograph_advantage:

**Comment by tsop14** [comment]
- Creates on text box on the left for free search. 
- In this search box you can select, paste and delete as normal. 
- Moves all chips to the left. 
chips deleting


- Removes the 'any' chip. 
- When you create a chip with a field that is not in the valid list, it returns it to text
chips

! You cannot select the text and the chips and start typing again.

File: kahuna/public/js/components/gr-chips/gr-filter-chooser-chip.js [hunk 1]
```diff
@@ -52,19 +52,16 @@ grFilterChooserChip.controller('grFilterChooserChipCtrl', function() {
...
+                // Not a key, turn it back to text search
+                $grChipsCtrl.focusEndOfFirstChip();
+                chip.value = "FIELD NOT FOUND";
+                $grFilterChooserChipCtrl.notValid = true;
+
+                $timeout(() => {
+                    $grFilterChooserChipCtrl.removeSelf();
+                    $grChipsCtrl.items[0].value += (" " + value);
+                }, 800);
+
```
Comment (NickPapacostas) [inline kahuna/public/js/components/gr-chips/gr-filter-chooser-chip.js]: why the delay?

**Comment by NickPapacostas** [comment]
:+1:  :ideograph_advantage:

File: kahuna/public/js/components/gr-chips/gr-filter-chooser-chip.js (hunk 1 above)
Comment (tsop14) [inline kahuna/public/js/components/gr-chips/gr-filter-chooser-chip.js]: Gives the user time to read the 'FIELD NOT FOUND' message (which is inside the chip) before said chip gets removed. 
Thoughts?

**Comment by tsop14** [comment]
Usages parsing fixes: 

NOW
screen shot 2016-03-21 at 17 03 26

BEFORE

screen shot 2016-03-21 at 17 02 42

Also creates one text 'chip' from all the text section of the parsed query and passes this through as the model value. Rather than 'normaliZing' in the gr-chips controller.

File: kahuna/public/js/search/structured-query/syntax.js [hunk 2]
```diff
@@ -39,7 +39,27 @@ export function structureQuery(query) {
...
+
+    const cleanStruct = [textChip].concat(filterChips);
+
+    function mergeTextValues(chips){
+        return chips.filter(e => e.type === 'text')
+            .map(e => e.value)
+            .join(' ');
+    }
+
+    return cleanStruct;
```
Comment (NickPapacostas) [inline kahuna/public/js/search/structured-query/syntax.js]: we could have just called it "struct" :disappointed:

**Comment by NickPapacostas** [comment]
:+1: :ideograph_advantage:

File: kahuna/public/js/search/structured-query/syntax.js (hunk 2 above)
Comment (tsop14) [inline kahuna/public/js/search/structured-query/syntax.js]: :boar:

**Pull request merged**